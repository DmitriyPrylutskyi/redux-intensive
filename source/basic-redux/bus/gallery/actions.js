// Types
import {
    SHOW_NEXT_IMAGE,
    SHOW_PREVIOUS_IMAGE,
    SHOW_SELECTED_IMAGE,
} from './types';

export const showNextImage = () => {
    return {
        type: SHOW_NEXT_IMAGE,
    };
};

export const showPreviousImage = () => {
    return {
        type: SHOW_PREVIOUS_IMAGE,
    };
};

export const showSelectedImage = (photoIndex) => {
    return {
        type:    SHOW_SELECTED_IMAGE,
        payload: photoIndex,
    };
};
