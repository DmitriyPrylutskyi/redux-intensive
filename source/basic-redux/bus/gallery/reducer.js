// Images
import image1 from '../../../theme/assets/images/1.jpeg';
import image2 from '../../../theme/assets/images/2.jpeg';
import image3 from '../../../theme/assets/images/3.jpeg';
import image4 from '../../../theme/assets/images/4.jpeg';

// Types
import {
    SHOW_NEXT_IMAGE,
    SHOW_PREVIOUS_IMAGE,
    SHOW_SELECTED_IMAGE,
} from './types';

const initialState = {
    images: [
        { id: '123', url: image1 },
        { id: '456', url: image2 },
        { id: '789', url: image3 },
        { id: '987', url: image4 },
    ],
    selectedImageIndex: 0,
};

export const galleryReducer = (state = initialState, action) => {
    switch (action.type) {
        case SHOW_NEXT_IMAGE:
            if (state.selectedImageIndex === state.images.length - 1) {
                return state;
            }

            return {
                ...state,
                selectedImageIndex: state.selectedImageIndex + 1,
            };

        case SHOW_PREVIOUS_IMAGE:
            if (state.selectedImageIndex === 0) {
                return state;
            }

            return {
                ...state,
                selectedImageIndex: state.selectedImageIndex - 1,
            };

        case SHOW_SELECTED_IMAGE:
            return {
                ...state,
                selectedImageIndex: action.payload,
            };

        default:
            return state;
    }
};
