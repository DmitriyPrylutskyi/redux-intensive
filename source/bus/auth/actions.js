// Types
import { types } from './types';

// Async
export const signupAsync = (userData) => {
    return {
        type:    types.SIGNUP_ASYNC,
        payload: userData,
    };
};

export const loginAsync = (credentials) => {
    return {
        type:    types.LOGIN_ASYNC,
        payload: credentials,
    };
};

export const authenticateAsync = () => {
    return {
        type: types.AUTHENTICATE_ASYNC,
    };
};

export const initializeAsync = () => {
    return {
        type: types.INITIALIZE_ASYNC,
    };
};

export const logoutAsync = () => {
    return {
        type: types.LOGOUT_ASYNC,
    };
};

// Sync
export const authenticate = () => {
    return {
        type: types.AUTHENTICATE,
    };
};

export const initialize = () => {
    return {
        type: types.INITIALIZE,
    };
};

export const logout = () => {
    return {
        type: types.LOGOUT,
    };
};
