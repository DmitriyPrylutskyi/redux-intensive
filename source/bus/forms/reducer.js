// Core
import { combineForms } from 'react-redux-form';

export const formsReducer = combineForms(
    {
        login: {
            email:    '',
            password: '',
            remember: false,
        },
        signup: {
            firstName: '',
            lastName:  '',
            email:     '',
            password:  '',
            invite:    'akI2Kynnvwic',
        },
        user: {
            profile: {
                firstName: '',
                lastName:  '',
                avatar:    [],
            },
            password: {
                oldPassword: '',
                newPassword: '',
            },
        },
        feed: {
            composer: {
                comment: '',
            },
        },
    },
    'forms',
);
