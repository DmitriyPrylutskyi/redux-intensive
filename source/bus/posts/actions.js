// Types
import { types } from './types';

// Async
export const fetchPostsAsync = () => {
    return {
        type: types.FETCH_POSTS_ASYNC,
    };
};
export const likePostAsync = (postId) => {
    return {
        type:    types.LIKE_POST_ASYNC,
        payload: postId,
    };
};

export const unlikePostAsync = (postId) => {
    return {
        type:    types.UNLIKE_POST_ASYNC,
        payload: postId,
    };
};

// Sync
export const fillPosts = (posts) => {
    return {
        type:    types.FILL_POSTS,
        payload: posts,
    };
};

export const likePost = (likedPostData) => {
    return {
        type:    types.LIKE_POST,
        payload: likedPostData,
    };
};

export const unlikePost = (unlikedPostData) => {
    return {
        type:    types.UNLIKE_POST,
        payload: unlikedPostData,
    };
};
