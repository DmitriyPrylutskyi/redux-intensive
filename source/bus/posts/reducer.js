// Core
import { List, fromJS } from 'immutable';

// Types
import { types } from './types';

const initialState = List();

export const postsReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.FILL_POSTS:
            return fromJS(action.payload); // List(Map, Map, Map)

        default:
            return state;
    }
};
