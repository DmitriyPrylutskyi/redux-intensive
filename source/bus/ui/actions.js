// Types
import { types } from './types';

// Sync
export const startFetching = () => {
    return {
        type: types.START_FETCHING,
    };
};

export const stopFetching = () => {
    return {
        type: types.STOP_FETCHING,
    };
};

export const setOnlineState = () => {
    return {
        type: types.SET_ONLINE_STATE,
    };
};

export const setOfflineState = () => {
    return {
        type: types.SET_OFFLINE_STATE,
    };
};
