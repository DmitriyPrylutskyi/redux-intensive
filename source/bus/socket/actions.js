// Instruments
import { socket } from '../../init/socket';
import { setOnlineState, setOfflineState } from '../ui/actions';
import { createPost, removePost, likePost, unlikePost } from '../posts/actions';
import { Map } from 'immutable';

// Api
import { api } from '../../API';

export const listenConnection = () => {
    return (dispatch) => {
        socket.on('connect', () => {
            dispatch(setOnlineState());
        });

        socket.on('disconnect', () => {
            dispatch(setOfflineState());
        });
    };
};

export const listenPosts = () => {
    return (dispatch, getState) => {
        socket.on('like', async (event) => {
            const response = JSON.parse(event);
            if (response.meta.action === 'dislike') {
                dispatch(unlikePost(response.data));
            } else {
                const { firstName, lastName } = await api.users.getById.call(
                    api,
                    response.data.userId,
                );

                const liker = Map({
                    firstName,
                    lastName,
                    id: response.data.userId,
                });

                dispatch(
                    likePost({
                        postId: response.data.postId,
                        liker,
                    }),
                );
            }
        });

        socket.on('create', (event) => {
            const response = JSON.parse(event);
            dispatch(createPost(response.data));
        });

        socket.on('remove', (event) => {
            const response = JSON.parse(event);
            dispatch(removePost(response.data));
        });
    };
};
