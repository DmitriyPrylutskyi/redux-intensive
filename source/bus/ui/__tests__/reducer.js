// Reducer
import { uiReducer } from '../reducer';

// Actions
import * as uiActions from '../actions';

describe('ui reducer:', () => {
    test('should return state by default', () => {
        expect(uiReducer(void 0, { type: 'TEST_ACTION' })).toMatchSnapshot();
    });
});
