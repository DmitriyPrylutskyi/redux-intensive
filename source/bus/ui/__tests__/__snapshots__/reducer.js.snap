// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui reducer: should return state by default 1`] = `
Immutable.Map {
  "isFetching": false,
  "isOnline": false,
}
`;
